pkg/api/types.go

```diff
@@ -2070,10 +2070,83 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// ComponentType constants define the known types of components.
+// Each type may have multiple instances.
+// Each instance's name is prefixed by the type, with a random suffix. See SimpleNameGenerator.
```

Comment:
I don't have all the api impls memorized, so take it easy.
The strategy is here: https://github.com/mesosphere/kubernetes/blob/component-registration/pkg/registry/component/strategy.go#L54-L83
If what you say is true, does it just explode if a CREATE of a new pod has a name of an existing pod?